packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,614 @@
+    if (
+      (this.state.isLegendSelected === false ||
+        (this.state.isLegendSelected && this.state.selectedLegendTitle === point.legend)) &&
+      this._calloutAnchorPoint !== point
+    ) {
+      this._calloutAnchorPoint = point;
+      this.setState({
+        refSelected: mouseEvent,
+        isCalloutVisible: true,
+        dataForHoverCard: point.x,
+        selectedLegendTitle: point.legend!,
+        color: point.color || color,
+        // To display callout value, if no callout value given, taking given point.x value as a string.
+        xCalloutValue: point.yAxisCalloutData! || point.y.toString(),
+        yCalloutValue: point.xAxisCalloutData || point.x.toString(),
+        dataPointCalloutProps: point,
+        activeXdataPoint: point.x,
+        YValueHover,
+        hoverXValue,
+        callOutAccessibilityData: point.callOutAccessibilityData,
+      });
+    }
+  }
+
+  private _onBarLeave = (): void => {
+    /**/
+  };
+
+  private _handleChartMouseLeave = (): void => {
+    this._calloutAnchorPoint = null;
+    this.setState({
+      isCalloutVisible: false,
+      activeXdataPoint: null,
+      YValueHover: [],
+      hoverXValue: '',
+    });
+  };
+
+  private _onBarFocus = (
+    point: IHorizontalBarChartWithAxisDataPoint,
+    refArrayIndexNumber: number,
+    color: string,
+  ): void => {
+    if (
+      this.state.isLegendSelected === false ||
+      (this.state.isLegendSelected && this.state.selectedLegendTitle === point.legend)
+    ) {
+      const { YValueHover, hoverXValue } = this._getCalloutContentForBar(point);
+      this._refArray.forEach((obj: IRefArrayData, index: number) => {
+        if (refArrayIndexNumber === index) {
+          this.setState({
+            refSelected: obj.refElement,
+            isCalloutVisible: true,
+            selectedLegendTitle: point.legend!,
+            dataForHoverCard: point.x,
+            color: point.color || color,
+            xCalloutValue: point.yAxisCalloutData || point.y.toString(),
+            yCalloutValue: point.xAxisCalloutData! || point.x.toString(),
+            dataPointCalloutProps: point,
+            activeXdataPoint: point.x,
+            YValueHover,
+            hoverXValue,
+            callOutAccessibilityData: point.callOutAccessibilityData,
+          });
+        }
+      });
+    }
+  };
+
+  private _getScales = (
+    containerHeight: number,
+    containerWidth: number,
+    isNumericScale: boolean,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ): { xBarScale: any; yBarScale: any } => {
+    if (isNumericScale) {
+      const xMax = d3Max(this._points, (point: IHorizontalBarChartWithAxisDataPoint) => point.x as number)!;
+      const yMax = d3Max(this._points, (point: IHorizontalBarChartWithAxisDataPoint) => point.y as number)!;
+      const xBarScale = d3ScaleLinear()
+        .domain(this._isRtl ? [xMax, 0] : [0, xMax])
+        .nice()
```

Comment:
do we use nice in vertical bar chart as well?

Reply:
Yes